plugin-dell/dphandler/managers.go

```diff
@@ -168,3 +170,101 @@ func getInfoFromDevice(uri string, deviceDetails dpmodel.Device, ctx iris.Contex
 	ctx.StatusCode(resp.StatusCode)
 	ctx.Write([]byte(respData))
 }
+
+//VirtualMediaActions performs insert and eject virtual media operations on the device based on the request
+func VirtualMediaActions(ctx iris.Context) {
+	uri := ctx.Request().RequestURI
+	uri = replaceURI(uri)
+	var deviceDetails dpmodel.Device
+	//Get device details from request
+	err := ctx.ReadJSON(&deviceDetails)
+	if err != nil {
+		log.Error("While trying to collect data from request, got: " + err.Error())
+		ctx.StatusCode(http.StatusBadRequest)
+		ctx.WriteString("Error: bad request.")
+		return
+	}
+	device := &dputilities.RedfishDevice{
+		Host:     deviceDetails.Host,
+		Username: deviceDetails.Username,
+		Password: string(deviceDetails.Password),
+		PostBody: deviceDetails.PostBody,
+	}
+
+	// insert virtual media request modifications
+	if strings.Contains(uri, "VirtualMedia.InsertMedia") {
+		var reqBody dpmodel.VirtualMediaInsert
+		err = json.Unmarshal(device.PostBody, &reqBody)
+		if err != nil {
+			errMsg := "While unmarshaling the request to insert a virtual media, got: " + err.Error()
+			log.Error(errMsg)
+			ctx.StatusCode(http.StatusInternalServerError)
+			ctx.WriteString(errMsg)
+			return
+		}
+
+		device.PostBody, err = json.Marshal(reqBody)
```

Comment:
if you are not changing reqBody, why unmarshalling and marshalling again.

Reply:
This is done to make sure correct parameters come to plugin code

Reply:
As the params are sent elsewhere they will be verified there isnt it ? Even so why marshal this again ? We end up with these two operations on every request without a need.

Reply:
Addressed the comment